src/main/controllers/ledger-service.js

```diff
@@ -1,151 +1,100 @@
 		}
 
-		return { basePath: matchResult[1], index: parseInt(matchResult[2], 10) };
-	}
-
-	controller.prototype.getAccounts = async function(args) {
-		args = args || {};
-
-		let indexOffset = args.start || 0;
-		let accountsNo = args.quantity || 1;
-		let eth = await getEth();
-
-		let derivationPath = args.derivationPath || getDefaultDerivationPath();
-		const pathComponents = obtainPathComponentsFromDerivationPath(derivationPath);
-
-		let addresses = {};
-		for (let i = indexOffset; i < indexOffset + accountsNo; i += 1) {
-			let path = pathComponents.basePath + (pathComponents.index + i).toString();
-			let address = await eth.getAddress_async(path);
-			addresses[path] = address.address;
+		let message = (err.message || '').toLowerCase();
+		if (message.indexOf('denied by the user') != -1) {
+			return reject('DENIED_BY_THE_USER');
 		}
 
-		return addresses;
+		reject(errorsMap[message] || 'UNKNOWN_ERROR');
 	};
 
-	async function _signTransaction(txData, derivationPath) {
-		let eth = await getEth();
-
-		derivationPath = derivationPath || getDefaultDerivationPath();
-		const pathComponents = obtainPathComponentsFromDerivationPath(derivationPath);
-		let path = pathComponents.basePath + pathComponents.index;
-		let account = await eth.getAddress_async(path);
-		if (
-			!account ||
-			!account.address ||
-			account.address.toLowerCase() != txData.from.toLowerCase()
-		) {
-			throw new Error('Invalid address');
-		}
-
-		// Encode using ethereumjs-tx
-		const tx = new EthereumTx(txData);
-		const chainId = getNetworkId();
-
-		// Set the EIP155 bits
-		tx.raw[6] = Buffer.from([chainId]); // v
-		tx.raw[7] = Buffer.from([]); // r
-		tx.raw[8] = Buffer.from([]); // s
-
-		// Encode as hex-rlp for Ledger
-		const hex = tx.serialize().toString('hex');
-
-		// Pass to _ledger for signing
-		let result = await (await getEth()).signTransaction_async(derivationPath, hex);
-
-		// Store signature in transaction
-		tx.v = Buffer.from(result.v, 'hex');
-		tx.r = Buffer.from(result.r, 'hex');
-		tx.s = Buffer.from(result.s, 'hex');
-
-		// EIP155: v should be chain_id * 2 + {35, 36}
-		const signedChainId = Math.floor((tx.v[0] - 35) / 2);
-		if (signedChainId !== chainId) {
-			throw new Error('Invalid signature received. Please update your Ledger Nano S.');
-		}
-
-		// Return the signed raw transaction
-		return `0x${tx.serialize().toString('hex')}`;
-	}
-
-	controller.prototype.signTransaction = async args => {
+	async function _signTransaction(args) {
 		args.dataToSign.from = args.address;
-		let derivationPath = args.derivationPath;
-		return new Promise((resolve, reject) => {
-			let signPromise = _signTransaction(args.dataToSign, derivationPath);
+		let signPromise = electron.app.web3Service.waitForTicket({
+			method: 'signTransaction',
+			args: [args.dataToSign],
+			contractAddress: null,
+			contractMethod: null,
+			onceListenerName: null,
+			customWeb3: createWeb3()
+		});
 
+		return new Promise((resolve, reject) => {
```

Comment:
There's no need for this. There's clearly a promise immediately below this. Just return that one.